tests/integration/test_repeat.py

```diff
@@ -23,42 +22,183 @@
+def test_array_empty_repeats_valid(repeats):
+    res_np = np.repeat([], repeats)
+    res_num = num.repeat([], repeats)
+    assert np.array_equal(res_np, res_num)
+
+
+@pytest.mark.parametrize("repeats", ([3, 4], [1, 2, 3]))
+def test_array_empty_repeats_invalid(repeats):
+    # numpy raises:
+    # ValueError: operands could not be broadcast together with shape (0,) (2,)
+    # while cunumeric is pass with the result []
+    res_num = num.repeat([], repeats)
+    assert np.array_equal(res_num, [])
+
+
+@pytest.mark.parametrize("repeats", (-3, 0, 3, 4.7, [], [-3], [0], [3], [4.7]))
+def test_array_empty_axis_valid(repeats):
+    res_np = np.repeat([], repeats, axis=0)
+    res_num = num.repeat([], repeats, axis=0)
+    assert np.array_equal(res_np, res_num)
+
+
+@pytest.mark.parametrize("repeats", (-3, 0, 3, 4.7, [], [-3], [0], [3], [4.7]))
+def test_array_empty_axis_invalid(repeats):
+    with pytest.raises(ValueError):
+        num.repeat([], repeats, axis=1)
+
+
+@pytest.mark.parametrize("repeats", (-3, [-3]))
+def test_array_int_repeats_negative(repeats):
+    with pytest.raises(ValueError):
+        num.repeat(3, repeats)
+
+
+@pytest.mark.parametrize("repeats", (0, 3, 4.7, [0], [3], [4.7]))
+def test_array_int_repeats_valid(repeats):
+    res_np = np.repeat(3, repeats)
+    res_num = num.repeat(3, repeats)
+    assert np.array_equal(res_np, res_num)
+
+
+@pytest.mark.parametrize("repeats", ([], [1, 2]))
+def test_array_int_repeats_invalid(repeats):
+    msg = r"scalar"
+    with pytest.raises(ValueError, match=msg):
+        num.repeat(3, repeats)
+
+
+@pytest.mark.parametrize("repeats", (0, 3, 4.7, [0], [3], [4.7], [2, 3, 4]))
+def test_array_1d_repeats_valid(repeats):
+    anp = np.array([1, 2, 3])
+    res_np = np.repeat(anp, repeats)
+    res_num = num.repeat(anp, repeats)
+    assert np.array_equal(res_np, res_num)
+
+
+@pytest.mark.parametrize("repeats", ([], [2, 3]))
+def test_array_1d_repeats_invalid(repeats):
+    anp = np.array([1, 2, 3])
+    with pytest.raises(ValueError):
+        num.repeat(anp, repeats)
 
 
-def test_axis():
+@pytest.mark.parametrize("repeats", (0, [0], 3, 4.7, [3], [4.7]))
+def test_array_2d_repeats_valid(repeats):
+    anp = np.array([[1, 3], [2, 4]])
+    res_np = np.repeat(anp, repeats)
+    res_num = num.repeat(anp, repeats)
+    assert np.array_equal(res_np, res_num)
+
+
+@pytest.mark.parametrize("repeats", ([], [2, 3]))
+def test_array_2d_repeats_invalid(repeats):
+    anp = np.array([[1, 3], [2, 4]])
+    with pytest.raises(ValueError):
+        num.repeat(anp, repeats)
+
+
+@pytest.mark.skip()
+@pytest.mark.parametrize("arr", ([1, 2, 3], [[1, 3], [2, 4]]))
+@pytest.mark.parametrize("repeats", (-3, [-3]))
+def test_array_1d_repeats_fatal_error(arr, repeats):
+    anp = np.array(arr)
+    # numpy raises "ValueError: negative dimensions are not allowed"
+    # while cunumeric got "Fatal Python error: Aborted"
+    num.repeat(anp, repeats)
```

Comment:
can this be marked `xfail` instead of skip or does running the test at all crash the test run?

Reply:
This can not be marked as _xfail_, it will cause the entire test case to crash if marked as _xfail_. The subsequent test cases will not run.